packages/tailwindcss/src/ast.ts

```diff
@@ -149,6 +149,9 @@ export function walk(
         context,
         path,
         replaceWith(newNode) {
+          if (replacedNode) {
+            throw new Error('Cannot replace a node more than once')
+          }
```

Comment:
We could also silently ignore this but then we are just hiding a bug.

Maybe we could write a better error message here that this is a Tailwind internal bug? /cc @adamwathan

Reply:
Can we just not throw here? Is there something the user can do that triggers this or does it only signal a bug in Tailwind?

Reply:
Nothing the user can do and signals a bug in Tailwind for sure because we are the only ones using this API. 

Only issue about silently ignoring is that we wouldn't immediately know about an issue. And worse, maybe some APIs start relying on this odd behavior.

If we had thrown here than the bug you ran into would be obvious but would also break everything 🫣

Reply:
Just dropping in a suggestion here:
```
Cannot replace a node more than once. This is a bug in Tailwind CSS. Please open an issue with a reproduction at https://github.com/tailwindlabs/tailwindcss/issues
```

Reply:
I feel like there are probably a million places in the code where we could add conditionals like this if we really wanted right? Feels weird to me that this would be the first time we ever introduced a throw to tell us we made a mistake. I don’t think we’ve ever done this any other time we fixed a bug.

Reply:
Yeah makes sense. I silently bailed so at least we are not mutating _other_ AST nodes.